### PR TITLE
docs: add error handling section to streaming helper

### PR DESCRIPTION
**Pull request by sor4chi**
https://github.com/honojs/hono/pull/2262
https://github.com/honojs/hono/issues/2164

### CODE REVIEW

File: helpers/streaming.md [hunk 1]
```diff
@@ -76,3 +76,36 @@ app.get('/sse', async (c) => {
...
+    // Pipe a readable stream.
+    await stream.pipe(anotherReadableStream)
+  }, (err, stream) => {
+    stream.writeln('An error occurred!')
+    console.error(err)
+  })
+})
+```
+
+stream will be automatically closed after the callbacks are executed.
```
Comment (yusukebe) [inline helpers/streaming.md:103]: `The stream` is better than `stream`.

**Comment by yusukebe** [comment]
@sor4chi 

Thanks! I've commented.

**Comment by sor4chi** [comment]
fixed!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
Thanks!

**Pull request merged**